### PR TITLE
refactor validateSchema and Validation.test to ES2015

### PR DESCRIPTION
**Pull request by songawee**
**What kind of change does this PR introduce?**

refactor and documentation

**Did you add tests for your changes?**

current tests pass

**If relevant, link to documentation update:**

N/A

**Summary**

refactor validateSchema and Validation.test to ES2015

update documentation to use yarn instead of npm and clarify steps needed to start testing

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

File: lib/validateSchema.js [hunk 1]
```diff
@@ -12,16 +14,16 @@ require("ajv-keywords")(ajv, ["instanceof"]);
 
 function validateSchema(schema, options) {
 	if(Array.isArray(options)) {
-		var errors = options.map(validateObject.bind(this, schema));
-		errors.forEach(function(list, idx) {
+		const errors = options.map((options) => validateObject(schema, options));
+		errors.forEach((list, idx) => {
 			list.forEach(function applyPrefix(err) {
 				err.dataPath = "[" + idx + "]" + err.dataPath;
```
Comment (Kovensky) [inline lib/validateSchema.js]:
```js
`[${idx}]${err.dataPath}`
```

File: lib/validateSchema.js [hunk 2]
```diff
@@ -30,20 +32,20 @@ function validateSchema(schema, options) {
...
-	var newErrors = [];
-	errors.forEach(function(err) {
-		var dataPath = err.dataPath;
-		var children = [];
-		newErrors = newErrors.filter(function(oldError) {
-			if(oldError.dataPath.indexOf(dataPath) >= 0) {
+	let newErrors = [];
+	errors.forEach((err) => {
+		const dataPath = err.dataPath;
+		let children = [];
```
Comment (Kovensky) [inline lib/validateSchema.js]: `const`

File: lib/validateSchema.js [hunk 3]
```diff
@@ -30,20 +32,20 @@ function validateSchema(schema, options) {
...
+		newErrors = newErrors.filter((oldError) => {
+			if(oldError.dataPath.includes(dataPath)) {
 				if(oldError.children) {
-					oldError.children.forEach(function(child) {
+					oldError.children.forEach((child) => {
```
Comment (chicoxyzzy) [inline lib/validateSchema.js]: I believe this could be
```js
children = oldError.children.slice(0);
```

**Comment by TheLarkInn** [comment]
Hi @songawee!!!! Would you mind syncing any conflicts. Also if you need _any_ assistance finishing up the PR, let me know. Happy to help.

File: lib/validateSchema.js (hunk 2 above)
Comment (songawee) [inline lib/validateSchema.js]: It was recommended in an earlier comment to re-assign `children` on line 48 so I think this has to stay a `let`. thanks!

File: lib/validateSchema.js [hunk 4]
```diff
@@ -30,22 +32,20 @@ function validateSchema(schema, options) {
...
+	errors.forEach((err) => {
+		const dataPath = err.dataPath;
+		let children = [];
+		newErrors = newErrors.filter((oldError) => {
+			if(oldError.dataPath.includes(dataPath)) {
 				if(oldError.children) {
-					oldError.children.forEach(function(child) {
-						children.push(child);
-					});
+					children = children.concat(oldError.children.slice(0));
```
Comment (songawee) [inline lib/validateSchema.js]: there was previously a comment to make this line `children = oldError.children.slice(0);`, but I think that would only give the last iteration's errors. It looks like children is an accumulator across many children, so I just made it concat with the recommended way.

**Comment by songawee** [comment]
hey @TheLarkInn! Actually just sat down to address the comments. Let me know if you need anything else. Cheers.

**Comment by TheLarkInn** [comment]
Thank you.

**Comment by TheLarkInn** [comment]
Congrats on merging your first webpack/webpack Pull Request. 🙇

**Pull request merged**